### PR TITLE
Fixes #147: New Git Manual Fathead

### LINKED ISSUE #147: GitManual: Source seems to have been taken down
Just tried clicking the MoreAt link when searching for e.g. `git push` and got a 404. Maybe we need a new source?

------
IA Page:  http://duck.co/ia/view/git_manual

### PR DESCRIPTION
**Pull request by samskeller**
Fixes #147. The git manual fathead was previously internal and using a broken source. I've written a fetcher script and a parser script to pull data from https://git-scm.com/docs/ and then parse it into what I think is the correct format for a Fathead into an output.txt file. I wasn't able to find documentation about how to test a new Fathead, though, so while the fetcher script and parser script run just fine, I'm not sure if the output file is going to populate the actual answer properly. Happy to do more work on this if it needs it, just not sure how to know if it does!

This is for this Instant Answer: https://duck.co/ia/view/git_manual

### CODE REVIEW

**Comment by daxtheduck** [comment]
@samskeller Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by jdorweiler** [comment]
@samskeller sorry I just saw this :frowning: 
I'll take a look at it today.

File: share/fathead/git_manual/output.txt [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+git config [<file-option>] [type] [-z|--null] [--name-only] --get-regexp name_regex [value_regex]
+git config [<file-option>] [type] [-z|--null] --get-urlmatch name URL
+git config [<file-option>] --unset name [value_regex]
+git config [<file-option>] --unset-all name [value_regex]
+git config [<file-option>] --rename-section old_name new_name
+git config [<file-option>] --remove-section name
+git config [<file-option>] [-z|--null] [--name-only] -l | --list
+git config [<file-option>] --get-color name [default]
+git config [<file-option>] --get-colorbool name [stdout-is-tty]
+git config [<file-option>] -e | --edit</pre></code>	https://git-scm.com/docs//config.html
```
Comment (jdorweiler) [inline share/fathead/git_manual/output.txt]: The abstract will have to go on a single line.  You can replace the newlines with a `<br>` tag to format it for the front end.

Comment (jdorweiler) [inline share/fathead/git_manual/output.txt]: Actually can you make that a literal `\n` to represent newlines.  I fixed the first one manually and put it on beta.  Here's what the entry in output.txt should look like. 
```
git config      A                                                                               Get and set repository or global options <pre><code>git config [<file-option>] [type] [-z|--null] name [value [value_regex]]\ngit config [<file-option>] [type] --add name value\ngit config [<file-option>] [type] --replace-all name value [value_regex]\ngit config [<file-option>] [type] [-z|--null] --get name [value_regex]\ngit config [<file-option>] [type] [-z|--null] --get-all name [value_regex]\ngit config [<file-option>] [type] [-z|--null] [--name-only] --get-regexp name_regex [value_regex]\ngit config [<file-option>] [type] [-z|--null] --get-urlmatch name URL\ngit config [<file-option>] --unset name [value_regex]\ngit config [<file-option>] --unset-all name [value_regex]\ngit config [<file-option>] --rename-section old_name new_name\ngit config [<file-option>] --remove-section name\ngit config [<file-option>] [-z|--null] [--name-only] -l | --list\ngit config [<file-option>] --get-color name [default]\ngit config [<file-option>] --get-colorbool name [stdout-is-tty]\ngit config [<file-option>] -e | --edit</pre></code>  https://git-scm.com/docs//config.html
```
https://beta.duckduckgo.com/?q=git+config&t=canonical&ia=about
selection_300

**Comment by samskeller** [comment]
No problemo, @jdorweiler! I've just updated the pull request to use literal `\n` characters instead of newline characters in the command abstracts in the `parse.py` script. Also updated the corresponding `output.txt` file. Let me know how it looks.

**Comment by jdorweiler** [comment]
@samskeller looks great.  I installed it on beta so you can check it out.  Let me know what you think.   https://beta.duckduckgo.com/?q=git+init&t=h&ia=about

Just a few small things left:
- you can remove `lib/DDG/Fathead/GitManual.pm`.  We're not using those for fatheads anymore.
- we got rid of the `share` directory and moved everything over to `lib`.  Could you move your scripts over there too.

**Comment by jdorweiler** [comment]
@samskeller I was checking it out again and it looks like you have some extra tabs in the abstract field.  This is messing up the "more at" link.  For example, the entry for `git init` has an extra tab after `[--template=<template_directory>]\n`

**Comment by samskeller** [comment]
@jdorweiler I noticed those strange links too and was just about to ask about it! Thanks for the explanation. I've removed the `GitManual.pm file`, moved everything into the `lib` directory, and also updated the `parser.py` (and the corresponding `output.txt`) to clean out tabs from the abstracts

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=git commit)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/git_manual_mobile.png)

**Comment by jdorweiler** [comment]
@samskeller great, thanks for fixing this IA!

I put you down as the maintainer since you pretty much re-wrote the whole thing :smile:

**Comment by jdorweiler** [comment]
@samskeller I'll let you know when the new data goes live.

----
After your Instant Answer goes live for the World to use, you will be able to see [traffic information] (http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by samskeller** [comment]
@jdorweiler awesome! Thanks for all your help on this. Especially cool that you can see traffic information in the IA page!

**Comment by jdorweiler** [comment]
:cake: 
https://duckduckgo.com/?q=git+grep&ia=about

I didn't know `git grep` had so many options until now :smile:

**Comment by samskeller** [comment]
@jdorweiler yeah i've definitely learned a bit more about git from doing this issue!

also, sorry to bug you but was just curious -- i don't actually see traffic information on the instant answer page (https://duck.co/ia/view/git_manual) -- does the feature need to be live for a while before a graph shows up or something? thanks!

**Comment by jdorweiler** [comment]
@samskeller are you logged in on duck.co?  I see traffic on the page when I'm logged in.

**Comment by samskeller** [comment]
@jdorweiler yep -- it says "samskeller" in the top right corner of the page

**Comment by jdorweiler** [comment]
@samskeller how about now?

**Comment by samskeller** [comment]
@jdorweiler it works! awesome, thanks!

**Pull request merged**